src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingLeadingAsteriskCheck.java

```diff
@@ -0,0 +1,127 @@
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks if the javadoc has leading asterisk on each line or not.
+ * </p>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingLeadingAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * Some description here // OK
+ *  * Another line of description // OK
+ *    Another line of description // violation, missing leading asterisk at the start
+ *  *&#47;
+ * public void test() {}
+ * </pre>
+ *
+ * @since 6.0
+ *
+ */
+@StatelessCheck
+public class MissingLeadingAsteriskCheck extends AbstractCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISSING_ASTERISK = "javadoc.missing.asterisk";
+
+    /**
+     * The new line character that separates each line of javadoc.
+     */
+    private static final String COMMENT_SEPARATOR = "\n";
+
+    /**
+     * This character is present at the start of a javadoc.
+     */
+    private static final char JAVADOC_COMMENT_START = '*';
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.BLOCK_COMMENT_BEGIN,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final String commentContent = JavadocUtil.getBlockCommentContent(ast);
+        if (isJavaDocComment(commentContent) && !isInlineComment(commentContent)) {
+            for (String singleLine : commentContent.split(COMMENT_SEPARATOR)) {
+                if (!CommonUtil.isBlank(singleLine) && singleLine.trim().charAt(0) != '*') {
+                    log(ast, MSG_MISSING_ASTERISK);
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks, if the given comment is an inline comment or not.
+     * @param comment content of javadoc.
+     * @return true, if the given comment is an inline comment.
+     */
+    private static boolean isInlineComment(String comment) {
+        return comment.split(COMMENT_SEPARATOR).length == 1;
```

Comment:
`comment.split(COMMENT_SEPARATOR).length == 1;` it is better to look for next BLOCK_COMMENT_END token

Reply:
Sure

Reply:
So to check if a comment is inline, I will check if there are any new lines between `BLOCK_COMMENT_START` and `BLOCK_COMMENT_END` correct?

Reply:
I guess you meant that comment is **not** inline.
Yes.

Reply:
Done, updated this method.

Reply:
@strkkk I removed this method because we don't need it. As there are no newlines in an inline document no violations can occur from it. So it doesn't matter whether a comment is inline or not.